### DISCUSSION TITLE
How do I publish/render multiple reports within the same .qmd file?

### DISCUSSION THREAD

**Discussion by ming-cui** [quarto-dev/quarto-cli#7855] [Q&A] [answered]
### Description

I'm working on a large dataset housed in a .qmd file and have conducted data cleaning and preliminary analyses within it. My file structure includes two distinct level 1 headings: # Project 1 and # Project 2. Beneath each heading, I've documented my work, including both prose and code specific to each project.

I am exploring ways to render and publish these two sections separately, allowing for distinct audiences to view and interact with the content relevant to their respective projects. While I initially considered using headings as identifiers for rendering and publishing, I'm open to alternative suggestions and workflows.

I would greatly appreciate any guidance or insights for efficiently handling multiple reports within the same .qmd file. Thanks!

**Comment by cscheid** [comment, marked as answer]
The typical workflow for Quarto is to split multiple reports into multiple files.

I suspect you put them in a single file because of shared datasets or intermediate variables. That's a natural thing to do, but in this case, you are probably better served by a workflow that creates the intermediate results in a separate file, and then use (eg) analysis.qmd, project1.qmd, and project2.qmd.

If you're willing to use .ipynb notebooks, you can consider using a notebook to store the intermediate results of your analysis, and then use [notebook embeds](https://quarto.org/docs/authoring/notebook-embed.html).

Your use case (where analyses have shared data, computation, and dependencies) is one where I believe Quarto isn't great at yet. We have plans to improve on this, but not in the immediate future.

**Reply by ming-cui** [reply]
Thanks!
